brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -65,23 +78,63 @@ protected void init() {
             return HttpResponse.of(HttpStatus.NOT_FOUND);
         });
         sb.service("/foo", (ctx, req) -> HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"));
-        sb.service("/extra",
+        sb.service("/async", (ctx, req) -> asyncResponse(future ->
+            future.complete(HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"))));
+
+        sb.service("/exception", (ctx, req) -> {
+            // TODO: how do we set status 503 and also retain the cause's message?
+            throw new IllegalStateException("not ready");
```

Comment:
`throw HttpResponseException.of(HttpResponse.of(HttpStatus.SERVICE_UNAVAILABLE, "not ready"))` ?

Reply:
this doesn't set the exception message, only the http body (and that isn't used as the message..)

Reply:
usually there's a way to set status independent of the throwable, but this can work if we can control the message of the HttpResponseException (ex by adding a parameter) currently it has null message.

Reply:
Actually, just throwing an exception was the right way to do in this case. Sorry about the noise.

Reply:
but I do want to control the status code on error.. this helps show if a default of 500 isn't accidentally assumed.